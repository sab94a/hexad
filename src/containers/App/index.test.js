import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { App } from './';

const list = {
    items: [{ id: 1, title: 'Title' }, { id: 2, title: 'Title' }],
    loading: false
}

const init = jest.fn()

describe('Root App', () => {
    let component;

    it('Should render correctly', () => {
        component = shallow(<App list={ list } onInit={ init } />);
        expect(shallowToJson(component)).toMatchSnapshot();
    });

    it('Should call onInit callback on render', () => {
        expect(init.mock.calls.length).toBe(1);
    })
});
