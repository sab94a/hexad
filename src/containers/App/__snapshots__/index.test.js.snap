// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root App Should render correctly 1`] = `
<Memo(List)
  items={
    Array [
      Object {
        "id": 1,
        "title": "Title 1",
      },
      Object {
        "id": 2,
        "title": "Title 2",
      },
      Object {
        "id": 3,
        "title": "Title 3",
      },
    ]
  }
/>
`;
